src/lib/utils/useScreenDimensions.tsx

```diff
@@ -14,53 +11,30 @@ export interface ScreenDimensions {
 
 export interface ScreenDimensionsWithSafeAreas extends ScreenDimensions {
   safeAreaInsets: SafeAreaInsets
-}
-
-export const ScreenDimensionsContext = createContext<ScreenDimensionsWithSafeAreas>(
-  null as any /* STRICTNESS_MIGRATION */
-)
-
-function getCurrentDimensions(): ScreenDimensions {
-  const { width, height } = Dimensions.get("window")
-  return {
-    width,
-    height,
-    orientation: width > height ? "landscape" : "portrait",
-  }
-}
-
-export const ProvideScreenDimensions: React.FC = ({ children }) => {
-  const [dimensions, setDimensions] = useState<ScreenDimensions>(getCurrentDimensions())
-
-  useEffect(() => {
-    const onChange = () => {
-      setDimensions(getCurrentDimensions())
-    }
-    Dimensions.addEventListener("change", onChange)
-    return () => {
-      Dimensions.removeEventListener("change", onChange)
-    }
-  }, [])
-
-  return (
-    <SafeAreaProvider>
-      <SafeAreaConsumer>
-        {(safeAreaInsets) => {
-          safeAreaInsets = safeAreaInsets || { top: 20, bottom: 0, left: 0, right: 0 }
-          return (
-            <ScreenDimensionsContext.Provider value={{ ...dimensions, safeAreaInsets }}>
-              {children}
-            </ScreenDimensionsContext.Provider>
-          )
-        }}
-      </SafeAreaConsumer>
-    </SafeAreaProvider>
-  )
+  fullWidth: number
+  fullHeight: number
 }
 
 /**
  * Call during render to be notified whenever `screenDimensions` changes
  */
-export function useScreenDimensions() {
-  return useContext(ScreenDimensionsContext)
+export const useScreenDimensions = (): ScreenDimensionsWithSafeAreas => {
+  const insets = useSafeAreaInsets()
+  const frame = useSafeAreaFrame()
+
+  return {
+    safeAreaInsets: insets,
+    width: frame.width,
+    height: frame.height,
+
+    get fullWidth() {
+      return this.width + this.safeAreaInsets.left + this.safeAreaInsets.right
+    },
+    get fullHeight() {
+      return this.height + this.safeAreaInsets.top + this.safeAreaInsets.bottom
+    },
```

Comment:
I think we should rename these values to both make it more obvious what they are, and to maintain backwards compatibility

```suggestion
    safeAreaWidth: frame.width,
    safeAreaHeight: frame.height,

    get width() {
      return this.safeAreaWidth + this.safeAreaInsets.left + this.safeAreaInsets.right
    },
    get height() {
      return this.safeAreaHeight + this.safeAreaInsets.top + this.safeAreaInsets.bottom
    },
```

and also to maintain backwards compatibility. i noticed e.g. the fancy modal is messed up now because it's still using `height` which no longer includes the safe area

![image](https://user-images.githubusercontent.com/1242537/115537046-55013700-a292-11eb-93c9-c04d1c3982ab.png)

Reply:
ah. I'll fix this. good catch.

Reply:
Here's how it is now:
<img width="503" alt="Screenshot 2021-04-22 at 11 14 46" src="https://user-images.githubusercontent.com/100233/115697958-01f3b680-a35c-11eb-8eaf-ff038ae5590e.png">

Reply:
ah that's still wrong. Maybe it would be better to go back to having the `Dimensions.get('screen')`

Reply:
Huh, ok. Should be even higher, closer to the top? I'll take a look.